instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/instrumentation.rb

```diff
@@ -46,14 +47,42 @@ def require_dependencies
           require_relative 'messaging_helper'
         end
 
-        def add_plugin(*targets)
+        def add_plugins(*targets)
           targets.each { |klass| klass.add_plugin(AwsSdk::Plugin) }
         end
 
         def loaded_constants
+          if gem_version >= CURRENT_MAJOR_VERSION
+            load_v3_constants
+          else
+            load_legacy_constants
+          end
+        end
+
+        def load_v3_constants
+          ::Aws.constants.each_with_object([]) do |c, constants|
+            m = ::Aws.const_get(c)
+            next unless unloaded_service?(c, m)
+
+            begin
+              constants << m.const_get(:Client)
+            rescue StandardError => e
+              OpenTelemetry.logger.warn("Constant could not be loaded: #{e}")
+            end
+          end
+        end
+
+        def unloaded_service?(constant, service_module)
```

Comment:
Nit: the name is a little confusing - I think it should maybe be `loaded_service?`